release-content/0.14/migration-guides/12163_Migrate_from_LegacyColor_to_bevy_colorColor.md

```diff
@@ -1,22 +1,142 @@
+
+```rust
+// Before
+let mut color = Color::rgb(0.0, 0.0, 0.0);
+color.set_b(1.0);
+
+// After
+let color = Color::srgb(0.0, 0.0, 0.0);
+let srgba = Srgba {
+    blue: 1.0,
+    ..Srgba::from(color),
+};
+let color = Color::from(srgba);
+```
+
+`Color::hex` has been moved to `Srgba::hex`. Call `.into()` or construct a `Color::Srgba` variant manually to convert it.
+
+`Color::rgb_linear` and `Color::rgba_linear` have been renamed `Color::linear_rgb` and `Color::linear_rgba` to fit the naming scheme of the `LinearRgba` struct.
+
+`Color::as_linear_rgba_f32` and `Color::as_linear_rgba_u32` have been removed. Call `LinearRgba::to_f32_array` and `LinearRgba::to_u32` instead, converting if necessary.
+
+Several other color conversion methods to transform LCH or HSL colors into float arrays or `Vec` types have been removed. Please reimplement these externally or open a PR to re-add them if you found them particularly useful.
+
+Vector field arithmetic operations on `Color` (add, subtract, multiply and divide by a f32) have been removed. Instead, convert your colors into `LinearRgba` space and perform your operations explicitly there. This is particularly relevant when working with emissive or HDR colors, whose color channel values are routinely outside of the ordinary 0 to 1 range.
+
+#### Alpha
+
+Alpha, also known as transparency, used to be referred to by the letter `a`. It is now called by its full name within structs and methods.
+
+- `Color::set_a` and `Color::a` are now `Color::set_alpha` and `Color::alpha`. These are part of the new `Alpha` trait.
+- Additionally, `Color::is_fully_transparent` is now part of the `Alpha`.
+
+#### CSS Constants
+
+The various CSS color constants are no longer stored directly on `Color`. Instead, they’re defined in the `Srgba` color space, and accessed via `bevy::color::palettes::css`. Call `.into()` on them to convert them into a `Color` for quick debugging use, and consider using the much prettier `tailwind` palette for prototyping. Furthermore, the `LIME_GREEN` color has been renamed to `LIMEGREEN` to comply with the standard naming.
+
+```rust
+// Before
+let color = Color::BLUE;
+
+// After
+use bevy::color::palettes::css::BLUE;
+
+let color = BLUE;
+```
+
+#### Switch to `LinearRgba`
+
+`WireframeMaterial`, `ExtractedUiNode`, `ExtractedDirectionalLight`, `ExtractedPointLight`, `ExtractedSpotLight`, and `ExtractedSprite` now store a `LinearRgba` rather than a polymorphic `Color`. Furthermore, `Color` no longer implements `AsBindGroup`. You should store a `LinearRgba` instead to avoid conversion costs.
+
+#### When to use `Color`, `LinearRgba`, or a different color space
+
+There are a few recommendations for how to choose the right color representation:
+
+- If you are writing a game or some other executable, pick a color space and use it consistently throughout your project. Since you are the only one using your code, you can choose whichever space works best.
```

Comment:
I'm not sure this is good advice. Constructing colors in code (or in a data file) as HSV for example is perfectly compatible with then using them as linear rgba for lights, and sRGBA is still useful for 2D UI even when the rest of a 3D game is linear rgba.

Reply:
Yeah, I'm considering removing this section entirely. I don't think it fits in the migration guide.

Reply:
Yeah, IMO this should be cut.

Reply:
Removed in ab89f31bbf8072b3a445aa949b42ae6fe53325bf :)